beetsplug/deezer.py

```diff
@@ -75,14 +75,22 @@ def album_for_id(self, album_id):
         deezer_id = self._get_id("album", album_id, self.id_regex)
         if deezer_id is None:
             return None
-
-        album_data = requests.get(self.album_url + deezer_id).json()
-        if "error" in album_data:
-            self._log.debug(
-                f"Error fetching album {album_id}: "
-                f"{album_data['error']['message']}"
-            )
+        try:
+            album_data = requests.get(
+                self.album_url + deezer_id, timeout=10
+            ).json()
+        except (
+            requests.exceptions.ConnectionError,
+            requests.exceptions.RequestException,
+        ) as e:
+            self._log.debug("Error fetching album {}\n Error: {}", deezer_id, e)
             return None
+        else:
```

Comment:
This probably doesn't need to be an `else:` block, since control will already have returned from this function in the exceptional case.